AWS/asg-status.5m.sh

```diff
@@ -0,0 +1,123 @@
+#  <xbar.var>array(IGNORED_ASGS=""): List of ASG names not to display.</xbar.var>
+
+# AWS CLI Profile & Region
+AWS_PROFILE="default"
+REGION="us-east-1"
+
+# Define the ordered list of ASGs with their display names
+ORDERED_ASGS=(
+    "production-app-server-asg|Production App Servers"
+)
+
+# Define ASGs to ignore
+IGNORED_ASGS=(
+    "asg-to-ignore"
+    "other-asg-to-ignore"
+)
+
+# Fetch Auto Scaling Groups Data
+asg_data=$(/usr/local/bin/aws autoscaling describe-auto-scaling-groups --region "$REGION" --profile "$AWS_PROFILE" --output json)
+
+# Ensure AWS CLI returned data
+if [ -z "$asg_data" ]; then
+    echo "ASG: No Data"
+    exit 1
+fi
+
+# Extract Relevant Data
+total_instances=0
+total_healthy=0
+total_unhealthy=0
+declare -A ASG_OUTPUT
+
+# Parse JSON Data using jq
+while IFS=$'\t' read -r asg_name instance_count healthy_count; do
+    # Ignore ASGs with 0 instances
+    if [[ "$instance_count" -eq 0 ]]; then
+        continue
+    fi
+
+    unhealthy_count=$((instance_count - healthy_count))
+
+    # Update totals
+    total_instances=$((total_instances + instance_count))
+    total_healthy=$((total_healthy + healthy_count))
+    total_unhealthy=$((total_unhealthy + unhealthy_count))
+
+    # Calculate health percentage for the ASG
+    if [[ "$instance_count" -gt 0 ]]; then
+        health_percentage=$((100 * healthy_count / instance_count))
+    else
+        health_percentage=0
+    fi
+
+    
+
+    # Determine health icon
+    if [[ "$health_percentage" -lt 100 ]]; then
+        health_icon="⚠️"
+    else
+        health_icon="✅"
+    fi
+
+    # Store ASG data for manual ordering
+    ASG_OUTPUT[$asg_name]="$health_icon $asg_name: $health_percentage% ($instance_count) | href=https://us-east-1.console.aws.amazon.com/ec2/home?region=us-east-1#AutoScalingGroupDetails:id=$asg_name;view=instanceManagement\n-- Healthy: $healthy_count | color=green\n-- Unhealthy: $unhealthy_count | color=red"
+done < <(echo "$asg_data" | jq -r '
+  .AutoScalingGroups[]
+  | select(.Instances | length > 0)
+  | [.AutoScalingGroupName, (.Instances | length), (.Instances | map(select(.HealthStatus == "Healthy")) | length)]
+  | @tsv')
+
+# Correctly Calculate Overall Health
+overall_health=0
+if [[ "$total_instances" -gt 0 ]]; then
+    overall_health=$((100 * total_healthy / total_instances))
+fi
+
+# Determine warning emoji if health is below 100%
+health_status_icon=""
+if [[ "$overall_health" -lt 100 ]]; then
+    health_status_icon="⚠️"
+elif [[ "$overall_health" -eq 100 ]]; then
+    health_status_icon="✅"
+fi
+
+# Display in xbar Format
+echo "AWS ASG: $overall_health% ($total_instances) $health_status_icon"
+echo "---"
+
+# Track ASGs that have already been displayed
+declare -A DISPLAYED_ASGS
+
+
+# Display ASGs in manual order
+for entry in "${ORDERED_ASGS[@]}"; do
+    IFS='|' read -r asg_key asg_display <<< "$entry"
```

Comment:
Combined with the xbar variable use change above, need to split on `:`.

```suggestion
    IFS=':' read -r asg_key asg_display <<< "$entry"
```